modules/openedx/s3/buckets.tf

```diff
@@ -1,33 +1,52 @@
 resource "aws_s3_bucket" "edxapp" {
-  bucket = lower(join("-", [var.client_shortname, "edxapp", var.environment]))
+  bucket   = lower(join("-", [var.client_shortname, "edxapp", var.environment]))
+  provider = aws.s3
+}
+
+resource "aws_s3_bucket_acl" "edxapp_acl" {
+  bucket = aws_s3_bucket.edxapp.id
   acl    = "private"
+}
+
+resource "aws_s3_bucket_cors_configuration" "edxapp_cors" {
+  bucket = aws_s3_bucket.edxapp.id
 
   cors_rule {
     allowed_headers = ["*"]
     allowed_methods = ["PUT", "GET"]
     allowed_origins = ["*"]
   }
+}
 
+resource "aws_s3_bucket" "edxapp_tracking_logs" {
+  bucket   = lower(join("-", [var.client_shortname, "edxapp", var.environment, "tracking", "logs"]))
   provider = aws.s3
 }
 
-resource "aws_s3_bucket" "edxapp_tracking_logs" {
-  bucket = lower(join("-", [var.client_shortname, "edxapp", var.environment, "tracking", "logs"]))
+resource "aws_s3_bucket_acl" "edxapp_tracking_logs_acl" {
+  bucket = aws_s3_bucket.edxapp_tracking_logs.id
   acl    = "private"
+}
 
+resource "aws_s3_bucket" "edxapp_course_discovery" {
+  count    = var.enable_course_discovery ? 1 : 0
+  bucket   = lower(join("-", [var.client_shortname, "edxapp", var.environment, "course-discovery"]))
   provider = aws.s3
 }
 
-resource "aws_s3_bucket" "edxapp_course_discovery" {
-  count = var.enable_course_discovery ? 1 : 0
-  bucket = lower(join("-", [var.client_shortname, "edxapp", var.environment, "course-discovery"]))
-  acl = "private"
+resource "aws_s3_bucket_acl" "edxapp_course_discovery_acl" {
+  count  = length(aws_s3_bucket.edxapp_course_discovery)
+  bucket = aws_s3_bucket.edxapp_course_discovery[0].id
```

Comment:
What will happen when `count    = var.enable_course_discovery ? 1 : 0` count = 0, this statement will throw error right?

Reply:
@farhaanbukhsh, why would it fail? We're checking the length one line above as a condition to create/destroy this resource. 

Besides, I've deployed this with `var.enable_course_discovery = false`.

Reply:
Ahh! my bad :)